### PR TITLE
Use hooks

### PR DESCRIPTION
**Pull request by jquense**
this is clearly an improvement for all our context usage.

### CODE REVIEW

File: src/test/ButtonGroup.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        className,
+        baseClass,
+        size && `${prefix}-${size}`,
+        toggle && `${prefix}-toggle`,
+      )}
+    />
+  );
+});
+
+ButtonGroup.displayName = 'ButtonGroup';
```
Comment (jquense) [inline src/test/ButtonGroup.js]: TIL you can just do this and they appear as normal components in the devtools and tests

**Comment by jquense** [comment]
@taion this is done 👍

File: src/AbstractNavItem.js [hunk 2]
```diff
@@ -15,76 +16,68 @@ const propTypes = {
...
+          ? navContext.activeKey === navKey
+          : active;
+    }
+
+    if (props.role === 'tab') {
+      props.tabIndex = isActive ? tabIndex : -1;
+      props['aria-selected'] = isActive;
+    }
+
+    const handleOnclick = useEventCallback(e => {
```
Comment (taion) [inline src/AbstractNavItem.js]: for my benefit... what does `useEventCallback` do? i see the `useCommittedRef` thing but i don't immediately understand it

File: src/DropdownToggle.js [hunk 3]
```diff
@@ -82,4 +82,5 @@ class DropdownToggle extends React.Component {
   }
 }
 
+// Needs to be a class FTM, because it needs to accept a ref that can be used with findDOMNode
```
Comment (taion) [inline src/DropdownToggle.js]: why wouldn't a forwardRef into a functional component work?

Comment (jquense) [reply src/DropdownToggle.js]: somewhere the ref doesn't get passed through, so we are actually relying on findDOMNode. I'll fix it up tho after i prune react-overlays and popper gets some hooks

File: src/AbstractNavItem.js (hunk 2 above)
Comment (jquense) [reply src/AbstractNavItem.js]: it returns a function with a stable identity that doesn't need to be invalidated or changed if the inner handler isn't stable. it's a less flexible equivalent to 

```js
someMethod() {
  this.props.handleFoo()
}
```

Comment (taion) [reply src/AbstractNavItem.js]: neat/complicated!

**Comment by jjustus** [comment]
This PR currently breaks ie11 because of the dependency on @restart/hooks, which currently emits functions using arrow syntax.

**Pull request merged**